live/app/utils/answer-validators.js

```diff
@@ -0,0 +1,91 @@
+import _ from 'pix-live/utils/lodash-custom';
+
+// Base class for answers validators
+//
+// Usage:
+//   let validator = new AnswerValidator.validatorForChallenge('QCM', answerValue); // returns a QcmValidator instance
+//   validator.isValid();    // returns a boolean
+//   validator.errorMessage; // returns the validation error message (if any)
+export default class AnswerValidator {
+
+  // Factory function: returns a new validator for the given challenge and answer value.
+  static validatorForChallenge(challengeType, value) {
+    const validatorClasses = {
+      'QCUIMG':    QcuValidator,
+      'QCU':       QcuValidator,
+      'QRU':       QcuValidator,
+      'QCMIMG':    QcmValidator,
+      'QCM':       QcmValidator,
+      'QROC':      QrocValidator,
+      'QROCM':     QrocmValidator,
+      'QROCM-IND': QrocmValidator,
+      'QROCM-DEP': QrocmValidator
+    };
+    const validatorClass = validatorClasses[challengeType.toUpperCase()];
+    return new validatorClass(value);
+  }
+
+  constructor(value, errorMessage) {
+    this.value        = value;
+    this.errorMessage = errorMessage;
+  }
+
+  isValid() {
+    return !!this._validate();
```

Comment:
Oui : c'est une astuce pour transformer la valeur de retour de `this._validate()` en booléen. Comme ça même si `_validate()` renvoie une valeur (plutôt que `true` ou `false`), `isValid` renvoie une valeur strictement booléenne.

Bon, mais en vrai je pense que c'était juste de l'over-précaution de ma part : je vire ça :)